numpy/core/shape_base.py

```diff
@@ -204,19 +204,15 @@ def atleast_3d(*arys):
         return res
 
 
-def _arrays_for_stack_dispatcher(arrays, stacklevel=4):
-    if not hasattr(arrays, '__getitem__') and hasattr(arrays, '__iter__'):
-        warnings.warn('arrays to stack must be passed as a "sequence" type '
-                      'such as list or tuple. Support for non-sequence '
-                      'iterables such as generators is deprecated as of '
-                      'NumPy 1.16 and will raise an error in the future.',
-                      FutureWarning, stacklevel=stacklevel)
-        return ()
-    return arrays
+def _arrays_for_stack_dispatcher(arrays):
+    if not hasattr(arrays, "__getitem__"):
```

Comment:
I didn't really see a reason to check whether it isn't an iterator.  If it doesn't have an `__iter__` it seems OK to rely on `tuple(arrays)` failing.

Python is weird (IMO) about:
```
class obj:
    def __len__(self):
        return 3
    def __getitem__(self, x):
        return x
```
ignoring the length (and thus iterating forever trying to find the end), but that is not something we need to worry about here.